### PR TITLE
Service Catalog migration test

### PR DESCRIPTION
**Pull request by wozniakjan**
**Description**

Changes proposed in this pull request:

- introducing periodic test that will mimic svcat migration

**Related issue(s)**

test code: https://github.com/kyma-project/kyma/pull/11684

### CODE REVIEW

File: prow/config.yaml [hunk 1]
```diff
@@ -983,6 +983,34 @@ presets:
       - name: gardener-skr-kyma-dev-kubeconfig
         mountPath: "/gardener"
         readOnly: true
+  - labels:
+      preset-btp-operator-stage-credentials: "true"
```
Comment (wozniakjan) [inline prow/config.yaml:987]: for the static SM credentials

File: prow/jobs/kyma/skr-svcat-migration.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+          repo: kyma
+          path_alias: github.com/kyma-project/kyma
+          base_ref: main
+        - org: kyma-project
+          repo: test-infra
+          path_alias: github.com/kyma-project/test-infra
+          base_ref: main
+      spec:
+        containers:
+          - image: "wozniakjan/kyma-integration:123-k8s1.16"
```
Comment (wozniakjan) [inline prow/jobs/kyma/skr-svcat-migration.yaml:36]: this is only for debugging purposes until https://github.com/kyma-project/test-infra/pull/3932 merges

File: prow/scripts/skr-svcat-migration.bash [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+readonly SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+readonly TEST_INFRA_SOURCES_DIR="$(cd "${SCRIPT_DIR}/../../" && pwd)"
+
+# shellcheck source=prow/scripts/lib/log.sh
+source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/log.sh"
+# shellcheck source=prow/scripts/lib/utils.sh
+source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/lib/utils.sh"
+
+if [[ "${BUILD_TYPE}" == "pr" ]]; then
```
Comment (adamwalach) [inline prow/scripts/skr-svcat-migration.bash:15]: jobguard doesn't work for periodic jobs
I think this script can be shortened to one line:
`make -C tests/fast-integration ci-skr-svcat-migration`
This way we could put it directly in the entrypoint of the prowjob:
```
command: make
args:
- -C
- tests/fast-integration
- ci-skr-svcat-migration
```

Comment (wozniakjan) [reply prow/scripts/skr-svcat-migration.bash]: perfect, thanks. I was told to use skr-integration.sh as a template but your proposal looks much neater.
https://github.com/kyma-project/test-infra/blob/f4478be4f8beae7b1a7e714a99425285555550f0/prow/scripts/skr-integration.sh#L1-L24

**Comment by wozniakjan** [comment]
sample log from pjtester: https://status.build.kyma-project.io/log?container=test&id=1417819886263996416&job=wozniakjan_test_of_prowjob_skr-azure-svcat-migration-dev

**Comment by adamwalach** [comment]
/hold
please, un-hold when the other PR is ready

File: prow/scripts/skr-svcat-migration.bash [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env bash
+
+# Script to run SKR svcat migration
+
+set -euo pipefail
+make -C /home/prow/go/src/github.com/kyma-project/kyma/tests/fast-integration ci-skr-svcat-migration
```
Comment (Ressetkk) [inline prow/scripts/skr-svcat-migration.bash:6]: This script is unnecessary. You can remove it in favour of defining command directly in the ProwJob template.

File: templates/data/skr-svcat-migration-data.yaml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        to: ../prow/jobs/kyma/skr-svcat-migration.yaml
+    render:
+      - localSets:
+          jobConfig_default:
+            annotations:
+              pipeline.clusterprovisioning: keb
+              pipeline.installer: keb
+              pipeline.test: fast-integration
+            path_alias: github.com/kyma-project/kyma
+            command: "/home/prow/go/src/github.com/kyma-project/test-infra/prow/scripts/skr-svcat-migration.bash"
```
Comment (Ressetkk) [inline templates/data/skr-svcat-migration-data.yaml:13]: In favour of separate script for that you can define the script execution directly here.
```
command: "make"
args:
    - "-C"
    - "/home/prow/go/src/github.com/kyma-project/kyma/tests/fast-integration ci-skr-svcat-migration"
```

**Review by Ressetkk — CHANGES_REQUESTED** [review]

File: templates/data/skr-svcat-migration-data.yaml (hunk 5 above)
Comment (wozniakjan) [reply templates/data/skr-svcat-migration-data.yaml]: thanks, applied in: https://github.com/kyma-project/test-infra/pull/3913/commits/b912001dafd9abf6e158852502adaf07763cc66b
tested through https://status.build.kyma-project.io/log?container=test&id=1418513673479000064&job=wozniakjan_test_of_prowjob_skr-azure-svcat-migration-dev

**Review by Ressetkk — DISMISSED** [review]

**Comment by wozniakjan** [comment]
seems like `pre-main-test-infra-validate-generated-files` doesn't like this PR
```
2021/07/28 17:53:03 UTC [ERROR] Rendered job files does not match templates and the configuration:
2021/07/28 17:53:03 UTC [INFO]  M prow/jobs/kyma/skr-svcat-migration.yaml
    Run:
        go run cmd/rendertemplates/main.go --config ../../templates/config.yaml
    in the development/tools directory of the repository and commit changes.
    For more info read: /docs/prow/templates.md
```
oddly enough when I run it, there is no diff
```
$ pwd
/home/wozy/projects/go/src/github.com/kyma-project/test-infra/development/tools

$ go run cmd/rendertemplates/main.go --config ../../templates/config.yaml
...

$ git status
On branch svcat-migration-bck
Your branch is up to date with 'wozniakjan/svcat-migration-bck'.
nothing to commit, working tree clean
```
And I am pretty sure I used the `rendertemplates` to generate the `prow/jobs/kyma/skr-svcat-migration.yaml` in the first place. 

@adamwalach , @Ressetkk can you please advise what to do in this case?

**Review by Ressetkk — APPROVED** [review]

**Pull request merged**